#! /bin/sh /usr/share/dpatch/dpatch-run
## Parcel voice icon doesn't reflect disabled status if voice isn't used.dpatch by  <root@debian.example.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch by Henri Beauchamp

@DPATCH@

diff -urN linden/indra/newview/llstatusbar.cpp linden-patched/indra/newview/llstatusbar.cpp
--- linden/indra/newview/llstatusbar.cpp	2008-01-04 00:07:13.000000000 +0100
+++ linden-patched/indra/newview/llstatusbar.cpp	2008-01-07 22:38:45.000000000 +0100
@@ -388,26 +388,18 @@
 		childSetVisible("restrictpush", FALSE);
 	}
 
-	BOOL voice_enabled = gVoiceClient->voiceEnabled();
 	BOOL have_voice = parcel && parcel->getVoiceEnabled(); 
-	if (!voice_enabled)
+	if (have_voice)
 	{
 		childSetVisible("status_no_voice", FALSE);
 	}
 	else
 	{
-		if (have_voice)
-		{
-			childSetVisible("status_no_voice", FALSE);
-		}
-		else if (!have_voice)
-		{
-			childSetVisible("status_no_voice", TRUE);
-			childGetRect( "status_no_voice", buttonRect );
-			r.setOriginAndSize( x, y, buttonRect.getWidth(), buttonRect.getHeight());
-			childSetRect( "status_no_voice", r );
-			x += buttonRect.getWidth();
-		}
+		childSetVisible("status_no_voice", TRUE);
+		childGetRect( "status_no_voice", buttonRect );
+		r.setOriginAndSize( x, y, buttonRect.getWidth(), buttonRect.getHeight());
+		childSetRect( "status_no_voice", r );
+		x += buttonRect.getWidth();
 	}
 
 	BOOL canBuyLand = parcel
